superflore/generators/ebuild/run.py

```diff
@@ -69,33 +71,31 @@ def main():
     global overlay
     global preserve_existing
     global mode
-    if len(sys.argv) == 2:
-        arg1 = sys.argv[1].replace('--', '')
-
-        if arg1 == 'all' or arg1 == 'update':
-            mode = arg1
-            preserve_existing = arg1 == 'all'
-        else:
-            preserve_existing = False
-            if arg1 not in active_distros:
-                ros_overlay.error('Unknown ros distro "{0}"'.format(arg1))
-                ros_overlay.error('Exiting...')
-                sys.exit(1)
-            else:
-                msg = 'Regenerating all packages for distro "{0}"'.format(arg1)
-                ros_overlay.info(msg)
-                mode = arg1
-    elif len(sys.argv) >= 2:
-        ros_overlay.error('Invalid arguments!')
-        print_usage()
+
+    parser = argparse.ArgumentParser(
+        'Deploy ROS packages into Gentoo Linux.')
+    parser.add_argument(
+        '--distro', help='regenerate packages for the specified distro.', type=str)
+    parser.add_argument(
+        '--all', help='regenerate all packages in all distros.')
+    args = parser.parse_args(sys.argv[1:])
```

Comment:
Also, pretty sure I shouldn't need to do the `sys.argv[1:]`, but argparse complained about invalid args otherwise. Any idea why?